dotnet/src/Connectors/Connectors.OpenAI/Core/AzureSdk/OpenAIClientCore.cs

```diff
@@ -51,7 +59,60 @@ internal sealed class OpenAIClientCore : ClientCore
             options.AddPolicy(new AddHeaderRequestPolicy("OpenAI-Organization", organization!), HttpPipelinePosition.PerCall);
         }
 
-        this.Client = new OpenAIClient(apiKey, options);
+        // Accepts the endpoint if provided, otherwise uses the default OpenAI endpoint.
+        var clientEndpoint = endpoint ?? httpClient?.BaseAddress;
+
+        if (clientEndpoint is null)
+        {
+            Verify.NotNullOrWhiteSpace(apiKey); // For Public OpenAI Endpoint a key must be provided.
+            clientEndpoint = new Uri(PublicOpenAIEndpoint);
+        }
+        else if (clientEndpoint?.PathAndQuery == "/")
+        {
+            // Adds missing /v1 if not provided.
+            clientEndpoint = new Uri($"{endpoint}v1");
+        }
+
+        this.Client = new OpenAIClient(clientEndpoint, CreateDelegatedToken(apiKey ?? string.Empty), options);
+
+        this.UpdateOpenAIClientPrivateFields(endpoint, options, apiKey);
+    }
+
+    private void UpdateOpenAIClientPrivateFields(Uri? endpoint, OpenAIClientOptions options, string? apiKey = null)
+    {
+        var type = typeof(OpenAIClient);
+        if (endpoint?.Scheme == "http")
+        {
+            if (apiKey is not null)
+            {
+                throw new KernelException("To use an APIKey you must provide a TLS protected (https) endpoint");
+            }
+
+            // When using Non-HTTPS APIs (local deployments like LM Studio), authentication is not supported by OpenAIClient Pipeline implementation.
+            // The current only way is removing the authentication policy thru reflection, to be able to use SDK Azure Client against custom APIs.
+            // Otherwise the error bellow will happen:
+            // System.InvalidOperationException : Bearer token authentication is not permitted for non TLS protected (https) endpoints.
+
+            var fieldInfo = type.GetField("_pipeline", BindingFlags.NonPublic | BindingFlags.Instance);
+
+            // Change the value of the private field
+            fieldInfo?.SetValue(this.Client, HttpPipelineBuilder.Build(options,
+                Array.Empty<HttpPipelinePolicy>(),
+                Array.Empty<HttpPipelinePolicy>(),
+                new ResponseClassifier())
```

Comment:
We should not use private reflection to muck with other library internals.

Reply:
Yes, I had a bad feeling about it, that's why I pinged you 😊

Seems that we can use a Policy to override the Url, used this approach.